elastio ec2 backup ecs stopped task (#29)
elastio/contrib: elastio-ec2-backup_ecs_stopped_task/readme.md

# Elastio EC2 backup ECS stopped task

## Step 1 - Create ECR polices

> 1. From AWS console, select "IAM service" =>`Policies` =>'Create Policy'.

Reviewer: ```suggestion
1. From AWS console, select "IAM service" =>`Policies` =>"Create Policy".
```

> # Elastio ec2 backup ecs stopped task

Reviewer 2: In English, abbreviations must *always* be capitalized, there are no exceptions.

```suggestion
# Elastio EC2 backup ECS stopped task
```

1. Create ECR polices:
    1. From AWS console, select "IAM service" =>`Policies` =>'Create Policy'.
    2. Under the **Service** tab select `Elastic Container Registry`.
    3. Under the **Actions** tab select `All Elastic Container Registry actions (ecr:*)`.
    4. Under the **Resources** tab select `specific` =>`Add ARN`. Then choose the region, leave your account number and select "Any" for the Repository name.
    5. Enter 'ECR_FullAccess' as a policy name.

2. Roles
    1. From AWS console, select IAM service =>`Roles`.
    2. If you already have a role for ECS tasks, add the following permissions: 

>     ```AmazonECSTaskExecutionRolePolicy, SecretsManagerReadWrite, AmazonECS_FullAccess, ECR_FullAccess, ElastioFullAdmin``` or create a new role with these permissions.

Reviewer 2: This doesn't look right.  I second Ivan's request to lint this markdown.

# Elastio EC2 backup ECS stopped task

## Step 1 - Create ECR polices

1. From AWS console, select "IAM service" =>`Policies` =>'Create Policy'.

> 

Reviewer: I think it's not necessary to add a newline between items in list.

1. For Add tags (optional), specify custom tags to
    associate with the policy and then choose "Next": "Review".

1. For Role name, type `ecs_task_execution_role` and choose "Create role".

## Step 3 - Build and push a docker image

> You can find the latest version by following this link.
<https://gallery.ecr.aws/elastio-dev/elastio-cli>

1. Edit file contrib/elastio-ec2-backup-ecs-stopped-task/failed_ecs_handler_config.py
    and put data about topic, brokers, vault.

1. Create ECR repository.

1. Register your docker cli by getting the ecr public image with the command:

    ```docker
    aws ecr-public get-login-password --region us-east-1 | \
    docker login --username AWS --password-stdin public.ecr.aws
    ```

1. Open the ECR repository you have created earlier
    and press the button "View push commands".
    The screen will list four commands.
    Run them in your terminal to build the docker image.

## Step 4 - Create a Fargate Cluster

1. From the AWS console, select `Elastic Container Service`=>`Create Cluster`.

1. Under "Select" cluster template, select "networking only".

1. Enter your cluster name and press `Create`.

## Step 5 - Create an ECS Task

> 1. Select `Task Definitions` from the left-side menu. Then choose
>     `Create new task definition`.

Reviewer: ```suggestion
1. Select `Task Definitions` from the left-side menu. Then choose `Create new task definition`.
```

# Elastio EC2 backup ECS stopped task

## Step 1 - Create ECR polices

1. From AWS console, select "IAM service" => `Policies` => "Create Policy".
1. Under the **Service** tab select `Elastic Container Registry`.
1. Under the **Actions** tab select `All Elastic Container Registry actions (ecr:*)`.
1. Under the **Resources** tab select `specific` =>`Add ARN`.
    Then choose the region, leave your account number and
    select "Any" for the Repository name.
1. Enter 'ECR_FullAccess' as a policy name.

## Step 2 - Roles

1. From AWS console, select IAM service =>`Roles`.

> 1. If you already have a role for ECS tasks, add the following permissions:
> 
>     1. AmazonECSTaskExecutionRolePolicy,
>     1. SecretsManagerReadWrite,
>     1. AmazonECS_FullAccess,
>     1. ECR_FullAccess,
>     1. ElastioBackupAdmin

Reviewer 2: ```suggestion
1. If you already have a role for ECS tasks, add the following policies to the role:

    1. `AmazonECSTaskExecutionRolePolicy`
    1. `SecretsManagerReadWrite`
    1. `AmazonECS_FullAccess`
    1. `ECR_FullAccess`
    1. `ElastioBackupAdmin`
```

    or create a new role with these permissions.
Creating a new role follow these steps:

1. In the navigation pane, choose "Roles" =>"Create role".
1. In the "Select type of trusted entity section",
    choose "AWS service" => "Elastic Container Service".
1. In "Select your use case", select the "Elastic Container Service Task",
    then choose "Next": "Permissions".
1. In the "Attach permissions policies" section, search for

    1. AmazonECSTaskExecutionRolePolicy,
    1. SecretsManagerReadWrite,
    1. AmazonECS_FullAccess,
    1. ECR_FullAccess,

>     1. ElastioBackupAdmin

Reviewer 2: Format this as `code` as I did in the previous section.

    select "Policies", then choose "Next": "Tags".
1. For Add tags (optional), specify custom tags to
    associate with the policy and then choose "Next": "Review".
1. For Role name, type `ecs_task_execution_role` and choose "Create role".

## Step 3 - Build and push a docker image

> You can find the latest version by following this link.
<https://gallery.ecr.aws/elastio-dev/elastio-cli>

1. Edit file contrib/elastio-ec2-backup-ecs-stopped-task/failed_ecs_handler_config.py
    and put data about topic, brokers, vault.
1. Create ECR repository.

> 1. Register your docker cli by getting the ecr public image with the command:

Reviewer 2: Once again, abbreviations in English are *always* capitalized.  Please fix this throughout

```suggestion
1. Register your docker CLI by getting the ECR public image with the command:
```
